app/controllers/subscriptions_controller.rb

```diff
@@ -0,0 +1,42 @@
+class SubscriptionsController < ApplicationController
+
+    def index
+        render json: { message: 'Hashes of data will get converted to JSON' }
+    end
+
+    def create
+        subscription = Subscription.create(subscription_params)
+        if subscription.save
+            render json: subscription
+        else
+            render json: {error: "Sorry, please try again."}
+        end
```

Comment:
For the purpose of consistent exception handling, there is a custom exception handler module in `app/controllers/concerns` but for these errors to kick in, the way we write our controller matters a lot. That being said, stick to bang(!) operator methods like: `.create!`, `.find_by!`, etc, so appropriate exceptions can be raised and therefore handled appropriately and uniformly. Please refer to #3 and go through some of the comments I wrote to @tirthajyoti-ghosh in this regard.